lib/algos/list/delimiterOrphanDeleteMarker.js

```diff
@@ -1,45 +1,35 @@
-'use strict'; // eslint-disable-line strict
-const Delimiter = require('./delimiter').Delimiter;
-const VSConst = require('../../versioning/constants').VersioningConstants;
-const { inc, FILTER_ACCEPT, FILTER_END, SKIP_NONE } = require('./tools');
-const VID_SEP = VSConst.VersionId.Separator;
-const { DbPrefixes } = VSConst;
-
+const DelimiterVersions = require('./delimiterVersions').DelimiterVersions;
+const { FILTER_END } = require('./tools');
 const DELIMITER_TIMEOUT_MS = 10 * 1000; // 10s
 const TRIM_METADATA_MIN_BLOB_SIZE = 10000;
-
 /**
- * Handle object listing with parameters. This extends the base class Delimiter
+ * Handle object listing with parameters. This extends the base class DelimiterVersions
  * to return the orphan delete markers. Orphan delete markers are also
  * refered as expired object delete marker.
  * They are delete marker with zero noncurrent versions.
  */
-class DelimiterOrphanDeleteMarker extends Delimiter {
+class DelimiterOrphanDeleteMarker extends DelimiterVersions {
     /**
-     * Delimiter listing of non-current versions.
+     * Delimiter listing of orphan delete markers.
      * @param {Object}  parameters            - listing parameters
      * @param {String}  parameters.beforeDate - limit the response to keys older than beforeDate
      * @param {RequestLogger} logger          - The logger of the request
      * @param {String} [vFormat]              - versioning key format
      */
     constructor(parameters, logger, vFormat) {
         super(parameters, logger, vFormat);
-
+        this.keyMarker = parameters.marker;
+        this.nextKeyMarker = parameters.marker;
         this.beforeDate = parameters.beforeDate;
-
-        this.skipping = this.skippingV1;
-        this.genMDParams = this.genMDParamsV1;
-
         this.keyName = null;
-        this.staleDate = null;
         this.value = null;
-
         // used for monitoring
-        this.evaluatedKeys = 0;
+        this.scannedKeys = 0;
     }
 
-    skippingV1() {
-        return SKIP_NONE;
+    genMDParamsV0() {
+        this.start = Date.now();
```

Comment:
Same than for the other PR, shouldn't this also be done in a `genMDParamsV1()` method?

Reply:
`genMDParamsV1()` calls `genMDParamsV0()`.
There is a unit test that ensures `this.start` is always assigned for v0 and v1.

Reply:
OK, I recommend to add a comment then, because the implementation of `genMDParamsV1` is in the base class which makes it not obvious.